provider/resource_keycloak_users_permissions.go

```diff
@@ -0,0 +1,272 @@
+	return nil
+}
+
+func resourceKeycloakUsersPermissionsCreate(data *schema.ResourceData, meta interface{}) error {
+	return resourceKeycloakUsersPermissionsUpdate(data, meta)
+}
+
+func resourceKeycloakUsersPermissionsUpdate(data *schema.ResourceData, meta interface{}) error {
+	keycloakClient := meta.(*keycloak.KeycloakClient)
+
+	realmId := data.Get("realm_id").(string)
+
+	if data.Get("enabled").(bool) {
+		err := keycloakClient.EnableUsersPermissions(realmId)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := keycloakClient.DisableUsersPermissions(realmId)
+		if err != nil {
+			return err
+		}
+	}
+
+	viewScopePolicyId, ok := data.GetOkExists("view_scope_policy_id")
+	if ok && viewScopePolicyId != nil {
+		err := setUsersScopePermissionPolicy(keycloakClient, realmId, "view", viewScopePolicyId.(string))
+		if err != nil {
+			return err
+		}
+	}
+	manageScopePolicyId, ok := data.GetOkExists("manage_scope_policy_id")
+	if ok && manageScopePolicyId != "" {
+		err := setUsersScopePermissionPolicy(keycloakClient, realmId, "manage", manageScopePolicyId.(string))
+		if err != nil {
+			return err
+		}
+	}
+	mapRolesScopePolicyId, ok := data.GetOkExists("map_roles_scope_policy_id")
+	if ok && mapRolesScopePolicyId != "" {
+		err := setUsersScopePermissionPolicy(keycloakClient, realmId, "map-roles", mapRolesScopePolicyId.(string))
+		if err != nil {
+			return err
+		}
+	}
+	manageGroupMembershipScopePolicyId, ok := data.GetOkExists("manage_group_membership_scope_policy_id")
+	if ok && manageGroupMembershipScopePolicyId != "" {
+		err := setUsersScopePermissionPolicy(keycloakClient, realmId, "manage-group-membership", manageGroupMembershipScopePolicyId.(string))
+		if err != nil {
+			return err
+		}
+	}
+	impersonateScopePolicyId, ok := data.GetOkExists("impersonate_scope_policy_id")
+	if ok && impersonateScopePolicyId != "" {
+		err := setUsersScopePermissionPolicy(keycloakClient, realmId, "impersonate", impersonateScopePolicyId.(string))
+		if err != nil {
+			return err
+		}
+	}
+	userImpersonatedScopePolicyId, ok := data.GetOkExists("user_impersonated_scope_policy_id")
+	if ok && userImpersonatedScopePolicyId != "" {
+		err := setUsersScopePermissionPolicy(keycloakClient, realmId, "user-impersonated", userImpersonatedScopePolicyId.(string))
+		if err != nil {
+			return err
+		}
+	}
+
+	return resourceKeycloakUsersPermissionsRead(data, meta)
+}
+
+func resourceKeycloakUsersPermissionsRead(data *schema.ResourceData, meta interface{}) error {
+	keycloakClient := meta.(*keycloak.KeycloakClient)
+	realmId := data.Get("realm_id").(string)
+
+	usersPermissions, err := keycloakClient.GetUsersPermissions(realmId)
+	if err != nil {
+		return handleNotFoundError(err, data)
+	}
+
+	data.SetId(usersPermissions.RealmId)
+	data.Set("realm_id", usersPermissions.RealmId)
+
+	data.Set("enabled", usersPermissions.Enabled)
+
+	if !usersPermissions.Enabled {
+		log.Printf("[WARN] Removing resource with id %s from state as it no longer enabled", data.Id())
```

Comment:
I'm not sure I understand this, why do we want to remove this resource from state if its disabled?  Also, we aren't actually removing from state here.  You'll want to use `data.SetId("")` to do that.